File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        sumr = [0 for _ in range(n)]
        suml = [0 for _ in range(n)]
        sml = 0
        smr = 0
        for i in range(n):
            sml += nums[i]
            suml[i] = sml
        for i in range(n - 1, -1, -1):
            smr += nums[i]
            sumr[i] = smr
        sm = []
        sm.append(sumr[1] - (n - 1) * nums[0])
        for i in range(1,n-1):
            sm.append(sumr[i + 1] - (n - i - 1) * nums[i] + (i) * nums[i] - suml[i - 1])
        sm.append((n - 1) * nums[n - 1] - suml[n - 2])
        return sm
        

File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        int n=nums.size(),sum=0;
        vector<int> prefix,suffix(n),res(n);
        for(auto it:nums)
        {
            sum+=it;
            prefix.push_back(sum);
        }
        sum=0;
        for(int i=n-1;i>=0;i--)
        {
            sum+=nums[i];
            suffix[i]=sum;
        }
        for(int i=0;i<n;i++)
        {
            res[i]=(suffix[i]-nums[i]*(n-i))+(nums[i]*(i+1)-prefix[i]);
        }
    return res;

    }
};